modules/olm-disconnected-operatorhub-disconnected-mode.adoc

```diff
@@ -0,0 +1,289 @@
++
+----
+manifests/
+└── etcd
+    ├── 0.0.12
+    │   ├── clusterserviceversion.yaml
+    │   └── customresourcedefinition.yaml
+    └── package.yaml
+----
++
+If you see files already in this structure, you can skip this step. However, if
+you instead see only a single file called `bundle.yaml`, you must first break
+this file up to conform to the required structure.
++
+You must separate the CSV content under `data.clusterServiceVersion` (each file
+in the list), the CRD content under `data.customResourceDefinition` (each file
+in the list), and the package content under `data.Package` into their own files.
+
+.. For the CSV file creation, find the following lines in the `bundle.yaml` file:
++
+[source,yaml]
+----
+data:
+  clusterServiceVersions: |
+----
++
+Omit those lines, but save a new file consisting of the full CSV resource
+content beginning with the following lines, removing the prepended `-`
+character:
++
+[source,yaml]
+.Example `clusterserviceversion.yaml` file snippet
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+[...]
+----
+
+.. For the CRD file creation, find the following line in the `bundle.yaml` file:
++
+[source,yaml]
+----
+  customResourceDefinitions: |
+----
++
+Omit this line, but save new files consisting of each, full CRD resource content
+beginning with the following lines, removing the prepended `-` character:
++
+[source,yaml]
+.Example `customresourcedefinition.yaml` file snippet
+----
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+[...]
+----
++
+.. For the package file creation, find the following line in the `bundle.yaml`
+file:
++
+[source,yaml]
+----
+  packages: |
+----
++
+Omit this line, but save a new file consisting of the package content beginning
+with the following lines, removing the prepended `-` character, and ending with
+a `packageName` entry:
++
+[source,yaml]
+.Example `package.yaml` file
+----
+channels:
+- currentCSV: etcdoperator.v0.9.4
+  name: singlenamespace-alpha
+- currentCSV: etcdoperator.v0.9.4-clusterwide
+  name: clusterwide-alpha
+defaultChannel: singlenamespace-alpha
+packageName: etcd
+----
+
+. **Create an Operator catalog image.**
+
+.. Save the following to a Dockerfile, for example named
+`custom-registry.Dockerfile`:
++
+[source,go]
+----
+FROM registry.redhat.io/openshift4/ose-operator-registry:v4.2 AS builder
+
+COPY manifests manifests
+
+RUN /bin/initializer -o ./bundles.db
+
+FROM scratch
+
+COPY --from=builder /build/bundles.db /bundles.db
+COPY --from=builder /build/bin/registry-server /registry-server
+COPY --from=builder /bin/grpc_health_probe /bin/grpc_health_probe
+
+EXPOSE 50051
+
+ENTRYPOINT ["/registry-server"]
+
+CMD ["--database", "bundles.db"]
+----
+
+.. Use the `podman` command to create and tag the container image from the
+Dockerfile:
++
+----
+$ podman build -f custom-registry.Dockerfile \
```

Comment:
@adellape we need to add a note about setting up podman (to authenticate with the customer portal). 
Without this, the build will fail because of the registry being used.

Reply:
Since we can't link to modules from within a module, I had to add the link to the registry login procedure in the "Additional resources" at the bottom of the assembly:

http://file.rdu.redhat.com/~adellape/083019/olm_disconnected/applications/operators/olm-disconnected.html